### PR TITLE
[ADD] Autocomplete on partner

### PR DESCRIPTION
**Pull request by rrahir**
Description of the issue/feature this PR addresses:

Current behavior before PR:

Desired behavior after PR is merged:

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/partner_autocomplete/models/res_partner.py [hunk 1]
```diff
@@ -0,0 +1,309 @@
...
+_logger = logging.getLogger(__name__)
+
+PARTNER_REMOTE_URL = 'https://partner-autocomplete.odoo.com/iap/partner_autocomplete'
+
+class ResPartner(models.Model):
+    _name = 'res.partner'
+    _inherit = 'res.partner'
+
+    company_data_id = fields.Integer('Company database ID')
+    additional_info = fields.Char('Additional info')
```
Comment (antonylesuisse) [inline addons/partner_autocomplete/models/res_partner.py]: Remove

File: addons/partner_autocomplete/models/res_partner.py [hunk 2]
```diff
@@ -0,0 +1,309 @@
...
+        if suggestions:
+            results = []
+            for suggestion in suggestions:
+                results.append(self._format_preview_data(suggestion))
+            return results
+        else:
+            return []
+
+    @api.model
+    def enrich_company(self, company_domain, company_data_id, vat):
```
Comment (antonylesuisse) [inline addons/partner_autocomplete/models/res_partner.py]: Called by javascript

File: addons/partner_autocomplete/models/res_partner.py [hunk 3]
```diff
@@ -0,0 +1,309 @@
...
+        response, error = self._rpc_remote_api('enrich', {
+            'domain': company_domain,
+            'company_data_id': company_data_id,
+            'vat': vat,
+        })
+        if response and response.get('company_data'):
```
Comment (antonylesuisse) [inline addons/partner_autocomplete/models/res_partner.py]: d =  response.get('company_data'):
# map country and state to ids
d._map_country_and_sate_to_ids:
d['additional_info'] = self.ref('partner_autocmploete.info_template').render(d)

return

File: addons/partner_autocomplete/models/res_partner_autocomplete_sync.py [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+
+from odoo import api, fields, models
+
+
+class ResPartnerAutocompleteSync(models.Model):
+    _name = 'res.partner.autocomplete.sync'
+
+    partner_id = fields.Many2one('res.partner', string="Partner", ondelete='cascade')
+    sync_date = fields.Datetime('Sync datetime', default=False)
+    error = fields.Char('Error', default=False)
```
Comment (antonylesuisse) [inline addons/partner_autocomplete/models/res_partner_autocomplete_sync.py]: remove

File: addons/partner_autocomplete_bridge_address_extended/__manifest__.py [hunk 5]
```diff
@@ -0,0 +1,11 @@
+# -*- coding: utf-8 -*-
```
Comment (mreficent) [inline addons/partner_autocomplete_bridge_address_extended/__manifest__.py]: `partner_autocomplete_address_extended` is shorter and nicer than `partner_autocomplete_bridge_address_extended`

File: addons/partner_autocomplete/models/res_partner.py [hunk 6]
```diff
@@ -0,0 +1,180 @@
...
+                    partner.message_post(body=partner.additional_info)
+                    partner.write({'additional_info': False})
+
+        return partners
+
+    @api.multi
+    def write(self, values):
+        record = super(ResPartner, self).write(values)
+        if len(self) == 1:
+            for partner in self:
```
Comment (tivisse) [inline addons/partner_autocomplete/models/res_partner.py]: Much performance ! ;)

**Pull request merged**